app/component/navigation/MessageBar.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component, PropTypes } from 'react';
```

Comment:
Two minor things I would consider:
  -move (unread)message filtering function to store (or atleast to connectToStores part)
  -change read status through action (and not directly through the store) so that we get rid of the component->store dependency